standalone-metastore/metastore-rest-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogAdapter.java

```diff
@@ -133,82 +134,54 @@ enum HTTPMethod {
   }
 
   enum Route {
-    TOKENS(HTTPMethod.POST, "v1/oauth/tokens",
-            null, OAuthTokenResponse.class),
-    SEPARATE_AUTH_TOKENS_URI(HTTPMethod.POST, "https://auth-server.com/token",
-            null, OAuthTokenResponse.class),
-    CONFIG(HTTPMethod.GET, "v1/config",
-            null, ConfigResponse.class),
-    LIST_NAMESPACES(HTTPMethod.GET, "v1/namespaces",
-            null, ListNamespacesResponse.class),
-    CREATE_NAMESPACE(HTTPMethod.POST, "v1/namespaces",
-            CreateNamespaceRequest.class, CreateNamespaceResponse.class),
-    LOAD_NAMESPACE(HTTPMethod.GET, "v1/namespaces/{namespace}",
-            null, GetNamespaceResponse.class),
-    DROP_NAMESPACE(HTTPMethod.DELETE, "v1/namespaces/{namespace}"),
-    UPDATE_NAMESPACE(HTTPMethod.POST, "v1/namespaces/{namespace}/properties",
-            UpdateNamespacePropertiesRequest.class, UpdateNamespacePropertiesResponse.class),
-    LIST_TABLES(HTTPMethod.GET, "v1/namespaces/{namespace}/tables",
-            null, ListTablesResponse.class),
-    CREATE_TABLE(HTTPMethod.POST, "v1/namespaces/{namespace}/tables",
-            CreateTableRequest.class, LoadTableResponse.class),
-    LOAD_TABLE(HTTPMethod.GET, TABLES_PATH,
-            null, LoadTableResponse.class),
-    REGISTER_TABLE(HTTPMethod.POST, "v1/namespaces/{namespace}/register",
-            RegisterTableRequest.class, LoadTableResponse.class),
-    UPDATE_TABLE(HTTPMethod.POST, TABLES_PATH,
-            UpdateTableRequest.class, LoadTableResponse.class),
-    DROP_TABLE(HTTPMethod.DELETE, TABLES_PATH),
-    RENAME_TABLE(HTTPMethod.POST, "v1/tables/rename",
-            RenameTableRequest.class, null),
-    REPORT_METRICS(HTTPMethod.POST, "v1/namespaces/{namespace}/tables/{table}/metrics",
-            ReportMetricsRequest.class, null),
-    COMMIT_TRANSACTION(HTTPMethod.POST, "v1/transactions/commit",
-            CommitTransactionRequest.class, null),
-    LIST_VIEWS(HTTPMethod.GET, "v1/namespaces/{namespace}/views",
-            null, ListTablesResponse.class),
-    LOAD_VIEW(HTTPMethod.GET, VIEWS_PATH,
-            null, LoadViewResponse.class),
-    CREATE_VIEW(HTTPMethod.POST, "v1/namespaces/{namespace}/views",
-            CreateViewRequest.class, LoadViewResponse.class),
-    UPDATE_VIEW(HTTPMethod.POST, VIEWS_PATH,
-            UpdateTableRequest.class, LoadViewResponse.class),
-    RENAME_VIEW(HTTPMethod.POST, "v1/views/rename",
-            RenameTableRequest.class, null),
-    DROP_VIEW(HTTPMethod.DELETE, VIEWS_PATH);
+    TOKENS(HTTPMethod.POST, "v1/oauth/tokens", null),
```

Comment:
I copied & pasted 1.9.1 of the upstream and removed unused fields. When our `/v1/config` returns the list of endpoints, the original form, such as `v1/namespaces`, doesn't work. We need to be adaptable to the new form such as `/v1/{prefix}/namespaces`.
https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/ResourcePaths.java

Reply:
what is the role of {`prefix}` - catalogName?